src/main/java/com/zendesk/maxwell/row/RowMap.java

```diff
@@ -180,34 +181,69 @@ private String getPartitionKeyFallback(String partitionKeyFallback) {
 		JsonGenerator generator = jsonGeneratorThreadLocal.get();
-		generator.writeObjectFieldStart(jsonMapName); // start of jsonMapName: {
 
-		for ( String key: data.keySet() ) {
-			Object value = data.get(key);
+		if(outputConfig.encryptData && !outputConfig.encryptAll){
+			ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+			JsonGenerator obj = jsonFactory.createGenerator(outputStream);
 
-			if ( value == null && !includeNullField )
-				continue;
+			obj.writeStartObject();
+			for (String key : data.keySet()) {
+				Object value = data.get(key);
 
-			if ( value instanceof List ) { // sets come back from .asJSON as lists, and jackson can't deal with lists natively.
-				List stringList = (List) value;
+				if (value == null && !outputConfig.includesNulls)
+					continue;
 
-				generator.writeArrayFieldStart(key);
-				for ( Object s : stringList )  {
-					generator.writeObject(s);
+				if ( value instanceof List ) { // sets come back from .asJSON as lists, and jackson can't deal with lists natively.
+					List stringList = (List) value;
+
+					obj.writeArrayFieldStart(key);
+					for ( Object s : stringList )  {
+						obj.writeObject(s);
+					}
+					obj.writeEndArray();
+				} else if (value instanceof RawJSONString) {
+					// JSON column type, using binlog-connector's serializers.
+					obj.writeFieldName(key);
+					obj.writeRawValue(((RawJSONString) value).json);
+				} else {
+					obj.writeObjectField(key, value);
 				}
-				generator.writeEndArray();
-			} else if ( value instanceof RawJSONString ) {
-				// JSON column type, using binlog-connector's serializers.
-				generator.writeFieldName(key);
-				generator.writeRawValue(((RawJSONString) value).json);
-			} else {
-				generator.writeObjectField(key, value);
 			}
+			obj.writeEndObject();
+			obj.close();
+			outputStream.close();
+			generator.writeStringField(jsonMapName, RowEncrypt.encrypt(outputStream.toString(), outputConfig.encryption_key, outputConfig.secret_key));
 		}
+		else {
+			generator.writeObjectFieldStart(jsonMapName); // start of jsonMapName: {
+
+			for (String key : data.keySet()) {
+				Object value = data.get(key);
 
-		generator.writeEndObject(); // end of 'jsonMapName: { }'
+				if (value == null && !outputConfig.includesNulls)
+					continue;
+
+				if (value instanceof List) { // sets come back from .asJSON as lists, and jackson can't deal with lists natively.
+					List stringList = (List) value;
+
+					generator.writeArrayFieldStart(key);
+					for (Object s : stringList) {
+						generator.writeObject(s);
+					}
+					generator.writeEndArray();
+				} else if (value instanceof RawJSONString) {
+					// JSON column type, using binlog-connector's serializers.
+					generator.writeFieldName(key);
+					generator.writeRawValue(((RawJSONString) value).json);
+				} else {
+					generator.writeObjectField(key, value);
+				}
+			}
+
+			generator.writeEndObject(); // end of 'jsonMapName: { }'
+		}
```

Comment:
Sorry, my request wasn't quite right - I was thinking you could reuse `writeMapToJSON` and pass it through an encryption phase afterwards, without copy/pasting the implementation.

At the time I wasn't very familiar with this part of the codebase though - since it writes data as it goes, what you'd actually need to to is to introduce some indirection so that the same logic can be used for both modes (encrypted & plaintext). That's a bit more effort, but I think it's worthwhile so we don't get divergent behaviour between this duplicated code.

I've knocked up a commit which shows what I mean here: https://github.com/webpt/maxwell/compare/webpt:encryption...zendesk:tcuthbertson/encryption
(untested, but it should be close to working)